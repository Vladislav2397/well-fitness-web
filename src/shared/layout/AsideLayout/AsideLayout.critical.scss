.aside-layout {
    &__container {
        display: flex;
    }

    &__content {
        flex-grow: 1;
        height: 100px;
    }

    &__aside {
        border-left: 1px solid rgba(#8584a4, 0.2);
    }
}
