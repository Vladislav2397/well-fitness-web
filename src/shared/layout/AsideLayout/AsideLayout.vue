<template lang="pug">

.b-aside-layout
    .__container.container
        .__content
            slot
        .__aside
            slot(
                name="aside"
            )

</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'

export type AsideLayoutProps = {
    //
}

@Component
export default class AsideLayout extends Vue {
    //
}
</script>

<!-- <style lang="scss" src="./AsideLayout.critical.scss" /> -->
<!-- <style lang="scss" src="./AsideLayout.main.scss" /> -->
