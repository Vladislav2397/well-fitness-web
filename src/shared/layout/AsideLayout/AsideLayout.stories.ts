import { Story } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import AsideLayout, { type AsideLayoutProps } from './AsideLayout.vue'
import { defineControls } from '@/shared/lib/stories'

export default {
    title: 'undefined-path/AsideLayout',
    component: AsideLayout,
    argTypes: defineControls({
        //
    }),
}

const Template: Story<AsideLayoutProps> = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { AsideLayout },
    methods: {
        handle: action('clicked'),
    },
    template: `<AsideLayout v-bind="$props" @click="handle" />`,
})

export const Default = Template.bind({})
Default.args = {
    //
}
