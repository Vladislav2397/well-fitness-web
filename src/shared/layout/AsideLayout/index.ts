import AsideLayout, { type AsideLayoutProps } from './AsideLayout.vue'

export { AsideLayout, AsideLayoutProps }
