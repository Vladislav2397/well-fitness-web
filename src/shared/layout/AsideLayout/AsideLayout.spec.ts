import { shallowMount } from '@vue/test-utils'
import AsideLayout /*, { type AsideLayoutProps }*/ from './AsideLayout.vue'

describe('test AsideLayout', () => {
    test('snapshot', () => {
        const wrapper = shallowMount(AsideLayout, {
            propsData: {
                //
            } /* as AsideLayoutProps*/,
        })
        expect(wrapper).toMatchSnapshot()
    })
})
