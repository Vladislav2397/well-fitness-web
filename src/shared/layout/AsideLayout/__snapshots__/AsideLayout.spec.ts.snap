// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AsideLayout snapshot 1`] = `
<div class="b-aside-layout">
  <div class="__container container">
    <div class="__content"></div>
    <!---->
  </div>
</div>
`;
